### ISSUE TITLE
Recursive looking aliases lead to aliases coloured as unknown-token

### ISSUE THREAD

**Issue by NonLogicalDev** [zsh-users/zsh-syntax-highlighting#652]
Example: If one has this configuration:
```
alias rm=rmi
alias rmi='nocorrect rm -i'
alias cp=cpi
alias cpi='nocorrect cp -i'
```
`rm`, `rmi`, `cp`, and `cpi` will all be colored red (unknown-token)

This seems to be a regression from the following commit at least: (the earliest I know where this was not an issue)

+ [`d766243f7a2a920b01a405d8fd0b12cb18f6d5bc`](https://github.com/zsh-users/zsh-syntax-highlighting/commit/d766243f7a2a920b01a405d8fd0b12cb18f6d5bc?ts=4)

**Comment by danielshahaf** [comment]
Thanks for the report.  I've added a regression test in ce10f20e77791ae0bf1c561ff64ce10993e3fc50 and posted a PR: #652

**Comment by danielshahaf** [comment]
Fixed in f32d1704b1e97f3a82f96ffd8ed828db7d84c56d.
